clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -601,8 +608,22 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
             Map<String, ByteBuffer> groupAssignment = performAssignment(joinResponse.data().leader(), joinResponse.data().protocolName(),
                     joinResponse.data().members());
 
+            List<SyncGroupRequestData.SyncGroupRequestAssignment> groupAssignmentList = new ArrayList<>();
+            for (Map.Entry<String, ByteBuffer> assignment : groupAssignment.entrySet()) {
+                groupAssignmentList.add(new SyncGroupRequestData.SyncGroupRequestAssignment()
+                        .setMemberId(assignment.getKey())
+                        .setAssignment(assignment.getValue().array())
```

Comment:
Should we be a little careful calling `array()` here? It might be safer to use `Utils.toArray`. We should probably have been more careful accessing this ByteBuffer in the first place since it is provided by plugins. I think we have the same problem in `ConsumerCoordinator.metadata()`.

Reply:
Sounds good!